### PR TITLE
[KOGITO-222] Introducing to KogitoDataIndexService

### PR DESCRIPTION
**Pull request by ricardozanini**
See:
https://issues.jboss.org/browse/KOGITO-222

In this PR we introduce a new controller to deploy the Kogito Data Index service image:

1. Supports Kubernetes clusters
2. Creates a `deployment` with the `quay.io/kiegroup/kogito-data-index` image
3. Creates a `ConfigMap` with a path for users to upload their protobuf files
4. The Strimzi and Infinispan operators need to be installed in the cluster for this service to be deployed

TODO:

1. ~Writes documentation to setup Kafka and Infinispan infra structure to use with Kogito Data Index~
2. ~Inject Infinispan services to env vars.~
3. ~Inject Kafka services to env vars.~

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by ricardozanini** [comment]
Oh yeah baby:

```shell
$ oc describe kogitodataindex

Name:         kogito-data-index
Namespace:    kogito-owner
Labels:       <none>
Annotations:  <none>
API Version:  app.kiegroup.org/v1alpha1
Kind:         KogitoDataIndex
Metadata:
  Creation Timestamp:  2019-08-30T12:26:43Z
  Generation:          3
  Resource Version:    6244909
  Self Link:           /apis/app.kiegroup.org/v1alpha1/namespaces/kogito-owner/kogitodataindices/kogito-data-index
  UID:                 6cdc16a4-cb21-11e9-90ae-0a76b78ac33a
Spec:
  Cpu Limit:    
  Cpu Request:  
  Env:
  Image:           quay.io/ricardozanini/kogito-data-index:latest
  Memory Limit:    
  Memory Request:  
  Name:            kogito-data-index
  Replicas:        1
Status:
  Condition:
    Condition:          Provisioning
    Reason:             NewReplicaSetAvailable
  Dependencies Status:  All dependencies OK
  Deployment Status:
    Available Replicas:  1
    Conditions:
      Last Transition Time:  2019-08-30T12:26:43Z
      Last Update Time:      2019-08-30T12:27:08Z
      Message:               ReplicaSet "kogito-data-index-f57f9dcb6" has successfully progressed.
      Reason:                NewReplicaSetAvailable
      Status:                True
      Type:                  Progressing
      Last Transition Time:  2019-08-30T12:28:36Z
      Last Update Time:      2019-08-30T12:28:36Z
      Message:               Deployment has minimum availability.
      Reason:                MinimumReplicasAvailable
      Status:                True
      Type:                  Available
    Observed Generation:     3
    Ready Replicas:          1
    Replicas:                1
    Updated Replicas:        1
  Service Status:
    Load Balancer:
Events:  <none>
```

**Comment by ricardozanini** [comment]
Infinispan Operators still uses 9.x version. We'll hold on add this operator dependency.

**Comment by ricardozanini** [comment]
@xiezhang7 @bmozaffa @cristianonicolai  could you kindly review this PR? Since it's a huge implementation (introduction to a new controller), please take your time:

1. I focused on keeping the same architecture as for `KogitoApp` with smalls changes that I'll replicate and improve over time
2. The aim of this controller is to deploy a new image that we call Kogito Data Index. The details of this feature can be viewed in [here](https://github.com/kiegroup/kogito-runtimes/wiki/Data-Index-Service). It depends on Infinispan and Kafka that the user will have to deploy manually at this first time, but we plan to deploy it automatically in the next sprints: [KOGITO-246](https://issues.jboss.org/browse/KOGITO-246) and [KOGITO-94](https://issues.jboss.org/browse/KOGITO-94)
3. The CLI will deploy the Data Index to make things easier for users point of view: [KOGITO-252](https://issues.jboss.org/browse/KOGITO-252)
4. I also updated the README.md with instructions for use this new feature. I kindly ask for native speakers to review it for grammars errors.

I should work until late night today, so I can catch up with your TZ, just ping me on the chat.

Cheers!

File: deploy/crds/app_v1alpha1_kogitodataindex_cr.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  name: kogito-data-index
+spec:
+  # If not informed, these default values will be set for you
+  name: "kogito-data-index"
+  # environment variables to set in the runtime container. Example: JAVAOPTS: "-Dquarkus.log.level=DEBUG"
+  env: {}
+  # number of pods to be deployed
+  replicas: 1
+  # image to use for this deploy
+  image: "quay.io/ricardozanini/kogito-data-index:latest"
```
Comment (cristianonicolai) [inline deploy/crds/app_v1alpha1_kogitodataindex_cr.yaml]: Should this be _quay.io/kiegroup/_ ?

**Pull request merged**